### PR TITLE
Add support for dynamic tracepoint provider in lttng

### PR DESCRIPTION
**Pull request by ThadHouse**
I want to run a few more CI tests just to make sure it works correctly

### CODE REVIEW

File: defaults/CLog.cmake [hunk 1]
```diff
@@ -38,10 +43,20 @@ function(CLOG_GENERATE_TARGET)
         list(APPEND clogfiles ${ARG_CLOG_C_FILE})
     endforeach()
 
+
+
```
Comment (nibanks) [inline defaults/CLog.cmake:47]: Do you have the same newline disease as Chris?

File: examples/clogsample/CMakeLists.txt [hunk 2]
```diff
@@ -16,3 +16,17 @@ CLOG_GENERATE_TARGET(CLOGSAMPLE_LIB STATIC ${SOURCES})
...
+target_link_libraries(clogsample ${LTTNGUST_LIBRARIES})
+
+CLOG_GENERATE_TARGET(CLOGSAMPLE_DYNLIB SHARED ${SOURCES})
+
+add_executable (clogsampledynamictp ${SOURCES})
+target_link_libraries(clogsampledynamictp CLOGSAMPLE_DYNLIB)
+
+add_library (clogsampletracepointprovider SHARED $<TARGET_OBJECTS:CLOGSAMPLE_DYNLIB.provider>)
+target_link_libraries(clogsampletracepointprovider ${LTTNGUST_LIBRARIES})
+endif()
```
Comment (nibanks) [inline examples/clogsample/CMakeLists.txt:32]:
```suggestion
add_library (clogsampletracepointprovider SHARED $<TARGET_OBJECTS:CLOGSAMPLE_DYNLIB.provider>)
target_link_libraries(clogsampletracepointprovider ${LTTNGUST_LIBRARIES})

endif()
```

File: defaults/CLog.cmake (hunk 1 above)
Comment (ThadHouse) [reply defaults/CLog.cmake:47]: Yes 👍

File: .azure/azure-pipelines.ci.yml [hunk 3]
```diff
@@ -77,6 +77,12 @@ stages:
         artifact: ToolNuPkg
         path: $(System.DefaultWorkingDirectory)/nupkg
     - pwsh: |
+        if ($IsLinux) {
+          sudo apt-add-repository ppa:lttng/stable-2.10
```
Comment (nibanks) [inline .azure/azure-pipelines.ci.yml:81]:
```suggestion
          sudo apt-add-repository ppa:lttng/stable-2.11
```
2.11 is the latest

File: examples/clogsample/CMakeLists.txt [hunk 4]
```diff
@@ -16,3 +16,18 @@ CLOG_GENERATE_TARGET(CLOGSAMPLE_LIB STATIC ${SOURCES})
...
+    include(FindLTTngUST)
+    target_link_libraries(clogsample ${LTTNGUST_LIBRARIES})
+
+    CLOG_GENERATE_TARGET(CLOGSAMPLE_DYNLIB SHARED ${SOURCES})
+
+    add_executable (clogsampledynamictp ${SOURCES})
+    target_link_libraries(clogsampledynamictp CLOGSAMPLE_DYNLIB)
+
+    add_library (clogsampletracepointprovider SHARED $<TARGET_OBJECTS:CLOGSAMPLE_DYNLIB.provider>)
+    target_link_libraries(clogsampletracepointprovider ${LTTNGUST_LIBRARIES})
```
Comment (nibanks) [inline examples/clogsample/CMakeLists.txt:31]:
```suggestion
    add_executable(clogsampledynamictp ${SOURCES})
    target_link_libraries(clogsampledynamictp CLOGSAMPLE_DYNLIB)

    add_library(clogsampletracepointprovider SHARED $<TARGET_OBJECTS:CLOGSAMPLE_DYNLIB.provider>)
    target_link_libraries(clogsampletracepointprovider ${LTTNGUST_LIBRARIES})
```

**Comment by ThadHouse** [comment]
Why'd this close? Do I need to recreate it against master?

**Comment by nibanks** [comment]
I believe he might have merged it in manually.

**Comment by ThadHouse** [comment]
Nope. I'll just open another copy against master.

**Pull request closed without merging**